core/server/src/main/java/alluxio/worker/netty/DataServerReadHandler.java

```diff
@@ -208,7 +210,8 @@ private long remainingToQueue() {
    * @return the number of bytes remaining to flush. Return 0 if it is cancelled
    */
   private long remainingToWrite() {
-    return mRequest.end() - mPosToWrite;
+    ReadRequestInternal request = mRequest;
```

Comment:
similarly, it is volatile. we need to make a copy of the ptr to make sure it is not null when it is used later.